### PR TITLE
Minor changes to publisher bool.

### PR DESCRIPTION
**Pull request by boks1971**
Some comments inline. Viewing with white space hiding makes it easier to read the diff.

### CODE REVIEW

File: pkg/rtc/participant.go [hunk 1]
```diff
@@ -1242,12 +1249,12 @@ func (p *ParticipantImpl) onMediaTrack(track *webrtc.TrackRemote, rtpReceiver *w
 		p.params.Logger.Warnw("no permission to publish mediaTrack", nil,
 			"source", publishedTrack.Source(),
 		)
+		p.removePublishedTrack(publishedTrack)
```
Comment (boks1971) [inline pkg/rtc/participant.go:1252]: Adding this in the publish path. That's the reason for splitting out the `remotePublishedTrack` method ☝️ .

I thought about adding the check before adding the track and not adding it. But, it is already published, so adding it and removing felt like the right flow and also the fact that things could race (i. e. permissions could be changing when a published track is being added, note that there are probably races still, for e.g. rapidly changing permissions).

File: pkg/rtc/participant.go [hunk 2]
```diff
@@ -1242,12 +1249,12 @@ func (p *ParticipantImpl) onMediaTrack(track *webrtc.TrackRemote, rtpReceiver *w
...
 		return
 	}
 
-	if !p.IsPublisher() {
-		p.setIsPublisher(true)
-	}
+	p.setIsPublisher(true)
```
Comment (boks1971) [inline pkg/rtc/participant.go:1256]: Just calling this without check as that function checks for change.

File: pkg/rtc/room.go [hunk 3]
```diff
@@ -898,6 +898,7 @@ func (r *Room) onTrackUnpublished(p types.LocalParticipant, track types.MediaTra
 }
 
 func (r *Room) onParticipantUpdate(p types.LocalParticipant) {
+	r.protoProxy.MarkDirty(false)
```
Comment (boks1971) [inline pkg/rtc/room.go:901]: Adding this as this gets called from `SetPermissions` which could change `isPublisher` bool.

Comment (davidzhao) [reply pkg/rtc/room.go:901]: if this is here, I think you can remove my previous changes when the track is published/unpublished.

**Review by davidzhao — APPROVED** [review]

File: pkg/rtc/room.go (hunk 3 above)
Comment (boks1971) [reply pkg/rtc/room.go:899]: Cool, addressed in https://github.com/livekit/livekit/pull/1880/commits/a46fd85bbefcda794214309a02f3133c0fb7ad87

**Pull request merged**